src/propagators/text_map_codec.js

```diff
@@ -68,12 +68,17 @@ export default class TextMapCodec {
       if (carrier.hasOwnProperty(key)) {
         let lowerKey = key.toLowerCase();
         if (lowerKey === this._contextKey) {
-          let decodedContext = SpanContext.fromString(this._decodeValue(carrier[key]));
+          let decodedContext = SpanContext.fromString(carrier[key]);
+          if (decodedContext !== null) { // trying to create context from split(":") worked ... no need to decode
```

Comment:
this can be simpler (one less `continue`) with
```
if (decodedContext == null) {
  decodedContext = SpanContext.fromString(this._decodeValue(carrier[key]));
}
```